src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -113,6 +113,21 @@ internal TarHeader(TarEntryFormat format, TarEntryType typeFlag, TarHeader other
         internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<string, string>> existing)
         {
             Debug.Assert(_ea == null);
+            foreach (KeyValuePair<string, string> kvp in existing)
```

Comment:
This is going to allocate an extra enumerator object.  It'd be better to construct the Dictionary and then enumerator the dictionary instead, as that won't add allocation to the success path.

Reply:
Fixed by bringing back the dictionary as it was before.